File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        res = s[0]
        vioCount = 0
        for c in s[1:]:
            if c == res[-1]:
                vioCount += 1
                if vioCount >= 2:
                    continue
            else:
                vioCount = 0
            res += c
        return res

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res = "";
        char lastChar = s[0];
        res += lastChar;
        int vioCount = 1;

        for(int i = 1; i < s.length(); i++){
            if(lastChar == s[i])
            {
                vioCount++;
                if(vioCount >= 3)
                {
                    continue;
                }
            }
            else
            {
                lastChar = s[i];
                vioCount = 1;
            }
            res += s[i];
        }
        return res;
    }
};